src/parser/mod.rs

```diff
@@ -628,13 +630,17 @@ fn parenthesized_expression<'a, 'b>(input: Input<'a, 'b>) -> AstResult<'a, 'b> {
 }
 
 parser!(variant loc =
-    name <- typename;
-    Ast::type_constructor(name, loc)
+    module_prefix <- delimited(typename, expect(Token::MemberAccess));
+    {
+        let (name, path) = module_prefix.split_last().unwrap();
+        Ast::type_constructor(path.to_owned(), name.to_string(), loc)
```

Comment:
Can we prevent the cloning here? Should be possible using `pop` instead of split_last